packages/@storefront/core/src/services/url.ts

```diff
@@ -124,7 +127,8 @@ class UrlService extends BaseService<UrlService.Options> {
   }
 
   buildUrlAndReplaceHistory = ({ state, route }: { state: Store.State, route: string }) => {
-    const url = this.beautifier.build(route, this.urlState[route](state));
+    const url = this.url || this.beautifier.build(route, this.urlState[route](state));
+    delete this.url;
```

Comment:
Do we want to be deleting this property, or just setting it to `null` or `undefined`? I would probably do the latter.

Reply:
@jmbtutor Are we concerned that `delete` may inadvertently remove a property on the prototype? If not, is there a reason to assign `undefined`?

Reply:
It's not that it would remove the property on the prototype (I don't think that's even possible without explicitly referring to the prototype), but as a matter of semantics, it's that we expect the `url` property to exist but to have some sentinel value. Because of that, I think we should explicitly set it to some sentinel value like `null` or `undefined` (or even `''` if that makes sense) instead of deleting it.

Reply:
@jmbtutor With that in mind, my gut says that setting it `undefined` is closer to the existing solution. That said, to what degree to we need to protect against `undefined` being redefined?